app/src/main/java/pl/futurecollars/invoicing/db/FileBasedDatabase.java

```diff
@@ -0,0 +1,96 @@
+package pl.futurecollars.invoicing.db;
+
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import lombok.AllArgsConstructor;
+import pl.futurecollars.invoicing.model.Invoice;
+import pl.futurecollars.invoicing.service.FileService;
+import pl.futurecollars.invoicing.service.IdService;
+import pl.futurecollars.invoicing.service.JsonService;
+
+@AllArgsConstructor
+public class FileBasedDatabase implements Database {
+
+  private final Path databasePath;
+  private final IdService idService;
+  private final JsonService jsonService;
+
+  @Override
+  public int save(Invoice invoice) {
+    try {
+      invoice.setId(idService.getNextIdAndSaveIncremented());
+      FileService.appendToFile(databasePath, jsonService.toJson(invoice));
+      return invoice.getId();
+    } catch (IOException exception) {
+      throw new RuntimeException("Failed to save the invoice");
```

Comment:
Jak już rzucasz wyjątek ponownie to wrzuć tam też informację o oryginalnym powodzie błędu, czyli:
` throw new RuntimeException("Failed to save the invoice, exception");`